streams/src/main/java/org/apache/kafka/streams/processor/internals/namedtopology/KafkaStreamsNamedTopologyWrapper.java

```diff
@@ -100,7 +100,7 @@ public void start(final NamedTopology initialTopology) {
     /**
      * Start up Streams with a collection of initial NamedTopologies (may be empty)
      */
-    public void start(final Collection<NamedTopology> initialTopologies) {
+    public synchronized void start(final Collection<NamedTopology> initialTopologies) {
```

Comment:
`super.start()` is already synchronized but we should just go ahead and synchronize at the first layer

Reply:
I took some time to understand why we want to synchronize here, as at the first sight it looks a bit unclear to me:

/* means caller -> callee */ 
inherited.start: synchronized, public -> 
addNamedTopology: unsynchronized, public, register topology metadata ->
completedFutureForUnstartedApp: synchronized, private, check state

removeNamedTopology: unsynchronized, public, unregister metadata topology ->
completedFutureForUnstartedApp: synchronized, private, check state

Register/unregister topology metadata is synchronized, and parent.start would modify state.

I think I understand now that it's because `addNamedTopology` is not synchronized, plus when we have multiple named topology we want to keep the state unchanged while adding them one-by-one. Is that the case? If yes maybe it's better to add such reasoning in the javadoc above.

Reply:
ack, will do